packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/remote_constraint_provider.dart

```diff
@@ -22,20 +31,128 @@ abstract class RemoteLoggingConstraintProvider {
 /// [LoggingConstraint] from an http endpoint periodically.
 /// {@endtemplate}
 class DefaultRemoteLoggingConstraintProvider
+    with AWSDebuggable, AWSLoggerMixin
     implements RemoteLoggingConstraintProvider {
   /// {@macro aws_logging_cloudwatch.default_remote_logging_constraint_provider}
   DefaultRemoteLoggingConstraintProvider({
     required DefaultRemoteConfiguration config,
     required AWSCredentialsProvider credentialsProvider,
   })  : _config = config,
-        _credentialsProvider = credentialsProvider;
+        _credentialsProvider = credentialsProvider {
+    _fetchAndCacheConstraintFromEndpoint(); // Fetch once immediately on initialize
+    _refreshConstraintPeriodically();
+  }
 
   final DefaultRemoteConfiguration _config;
   final AWSCredentialsProvider _credentialsProvider;
 
+  LoggingConstraint? _loggingConstraint;
+
+  /// The signer to sign the request.
+  static const _signer = AWSSigV4Signer();
+
+  // The timer to refresh the constraint periodically.
+  Timer? _timer;
+
+  /// Whether the periodic fetch is running.
+  bool isRunning = false;
+
+  /// Stops the periodic fetch.
+  void stop() {
+    _timer?.cancel();
+    isRunning = false;
+  }
+
   @override
-  // TODO(nikahsn): add implementation.
-  LoggingConstraint get loggingConstraint => throw UnimplementedError();
+  String get runtimeTypeName => 'DefaultRemoteLoggingConstraintProvider';
+
+  Future<void> _saveConstraintLocally(LoggingConstraint constraint) async {
+    await storage.saveConstraintLocally(constraint.toJson());
+  }
+
+  Future<void> _fetchAndCacheConstraintFromEndpoint() async {
+    try {
+      final constraint = await _fetchConstraintFromEndpoint();
+      if (constraint != null) {
+        _loggingConstraint = constraint;
+        await _saveConstraintLocally(constraint);
+      }
+      // ignore: avoid_catches_without_on_clauses
+    } catch (error) {
```

Comment:
It's a no-no to ignore these, they are trying to help you

https://dart.dev/language/error-handling

```suggestion
    } on Exception catch (error) {
```